### DISCUSSION TITLE
theme.json changes not available to tailwind after watch runs.

### DISCUSSION THREAD

**Discussion by victorcl** [gregsullivan/_tw#78] [Q&A] [answered]
Hi and  thanks for this great wordpress theme starter. 

I am having an issue which I haven't been able to sort out. When I make changes to the theme.json, watch gets triggered with "Rebuilding..." but these changes are not available to tailwind. 
But if I run "npm run dev" then the changes from theme.json are available to tailwing. But then if I make a change to any other file that triggers "watch" then I loose the theme.json changes again, they reset to the original values.
Any idea¿

Thanks.

**Comment by gregsullivan** [comment]
Hi Victor—Thanks for posting this!

In a funny bit of timing, I encountered a similar issue on Friday, and I had made a note to look into it further.

One question for you: When you run `npm run dev`, is the `watch` process still running?

My workaround over the weekend was to shut down the `watch` process and then to start it again, and that would pick up and persist the `theme.json` changes.

(This isn't meant to indicate an acceptance of this process as the new status quo; I'm going to put some time into figuring out why this stopped working and how to fix it. I'm not sure when I'll get to that, though.)

Once I hear back from you to confirm you're experiencing the same problem, I'll create an issue to track my progress.

**Reply by gregsullivan** [reply, marked as answer]
Having done significant testing, I think my advice for now is that you'll need to terminate and restart the `watch` process each time you modify the `theme.json` file.

I'm going to revisit this one after Oxide (Tailwind's upcoming Rust-based compiler) has been released.

**Reply by victorcl** [reply]
Hi Greg.

About your question _When you run npm run dev, is the watch process still running?_ is yes the `watch` process was still running when I ran `npm run dev`.

Thanks for the solution, it works fine for me,  if I restart the watch process, then the changes to theme.json persist.

**Reply by gregsullivan** [reply]
Perfect, thanks for letting me know! I've also added [an issue](https://github.com/gregsullivan/_tw/issues/79) to track efforts towards fixing this.
